File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/ProvincesService.java
package com.pinyougou.sellergoods.service;
import java.util.List;
import com.pinyougou.pojo.TbProvinces;							  
import entity.PageResult;
import entity.Result;
/**
 * provinces服务层接口
 * @author jieweili
 *
 */
public interface ProvincesService {

    /**
     * 返回全部列表(已弃置)
     * @return
     */
    List<TbProvinces> findAll();
 
    /**
     * 返回分页对象(过时)
     * @param pageNum
     * @param pageSize
     * @return
     */
	 
    PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 添加数据
     * @param provinces
     * @return
     */
    public Result saveProvinces(TbProvinces provinces);

    /**
     * 根据id查询一个对象
     * @param id
     * @return
     */
    TbProvinces getProvincesById(Integer id);

    /**
     * 修改对象
     * @param provinces
     * @return
     */
    Result updateProvinces(TbProvinces provinces);

    /**
     * 批量删除
     * @param idList
     * @return
     */
    Result removeProvinces(Integer[] idList);

    /**
     * 根据条件分页查询
     * @param provinces
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult listProvincesByExample(TbProvinces provinces, Integer pageNum, Integer pageSize);
 
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/provincesService.js
app.service('provincesService', function ($http) {
    //旧的查询方法(弃置)								   
    this.findPage = function (pageNum, pageSize) {
        return $http.get('../provinces/findPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize)
    };

    //查询方法,读取列表数据绑定到表单中
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../provinces/findPage.do?pageNum=' + pageNum + "&pageSize=" + pageSize, searchEntity)
    };

    //删除方法
    this.removeProvincesList = function (idList) {
        return $http.get('../provinces/removeProvinces.do?idList=' + idList)
    };

    //根据id查询
    this.findOne = function (id) {
        return $http.get('../provinces/findOne.do?id=' + id)
    };

    //修改品牌
    this.updateProvinces = function (provinces) {
        return $http.post('../provinces/updateProvinces.do', provinces)
    };

    //新增品牌
    this.saveProvinces = function (provinces) {
        return $http.post('../provinces/saveProvinces.do', provinces)
    };   
});
